src/contract.rs

```diff
@@ -86,10 +88,12 @@ pub fn handle<S: Storage, A: Api, Q: Querier>(
         // todo BurnFrom
         HandleMsg::Allowance /* todo make query? */ { spender, .. } => try_check_allowance(deps, env, spender),
         HandleMsg::Approve /* todo unspecified??? */ { spender, amount, .. } => try_approve(deps, env, &spender, amount),
+
+        // todo Send
+        HandleMsg::Swap { amount, network, destination, .. } => try_swap(deps, env, amount, network, destination),
     }
 }
-
-pub fn query<S: Storage, A: Api, Q: Querier>(deps: &Extern<S, A, Q>, msg: QueryMsg) -> QueryResult {
+pub fn authenticated_queries<S: Storage, A: Api, Q: Querier>(deps: &Extern<S, A, Q>, msg: &QueryMsg) -> QueryResult {
     let (address, key) = msg.get_validation_params();
```

Comment:
You can take `deps` and `msg` by value here with no problem, and then you won't need some other changes you made here.